src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,107 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo },
+      isFound: false
+    };
+  }
+
+  handleDivClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+    for (var file in this.props.reduxState.svgOptions) {
+      if (this.props.reduxState.svgOptions[file] === this.props.value) {
+        this.state.isFound = true;
+        this.state.index = file;
+      }
```

Comment:
perhaps look into array.find

Reply:
Its shorter, more readable and exits the loop as soon as a result is found which means its more efficient as well. Consider an array with a thousand records you would iterate through all 1000 even if the record was found on the first try.

Reply:
Dammit, I do not really know why I did not go through this function step by step and its really make sense now, I did not realised that the function was searching for one value here. thanks

Reply:
I thought of using this approach but ` this.state.isFound = this.props.svgOptions.find(function(option) {
      return option == this.props.value;
    });`  but since I also want the index of the Item , I just chose to put a break just after an item is found so it won't loop unnecessarily. I hope thats fine since

Reply:
so are you saying that there is no way to make this work with array.find, or array.findIndex, as it is the more readable and efficient way of doing it?

Reply:
There is a way to achieve it using those built in functions but I will have to loop twice , I do  agree with readability and not  with efficiency. I will use those built in function but it will be less efficient. `findIndex` and `find` are both loops.  Maybe I am wrong let me change it and do as you suggest that was my reasoning.

Reply:
I have decided to use findIndex then.